pkg/apis/pmemcsi/v1alpha1/deployment_types.go

```diff
@@ -111,6 +111,11 @@ type DeploymentStatus struct {
 // Deployment is the Schema for the deployments API
 // +kubebuilder:subresource:status
 // +kubebuilder:resource:path=deployments,scope=Namespaced
+// +kubebuilder:printcolumn:name="DeviceMode",type=string,JSONPath=`.spec.deviceMode`
+// +kubebuilder:printcolumn:name="NodeSelector",type=string,JSONPath=`.spec.nodeSelector`
+// +kubebuilder:printcolumn:name="Version",type=string,JSONPath=`.spec.image`
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=`.status.phase`
+// +kubebuilder:printcolumn:name="Age",type=data,JSONPath=`.metatdata.creationTimestamp`
```

Comment:
What is the effect of this?

"Version" is mapped to ".spec.image" here, which is not consistent with `AdditionalPrinterColumns` below.

Reply:
It was by mistake, I was initially thinking of showing only the image version and later decided to show the full image. Now I fixed this.

Reply:
I still don't understand why we have two different places for defining the "kubectl get" output. How are they related? Which one is used where? Do we need to maintain both manually?

Reply:
One is for making it work when installed the CRD generated using the kube-builder tool and the other is CRD installed from within the operator.

Reply:
Why do we have two ways of installing the CRD? When do we use which approach?

If this cannot be avoided, then at least cross-reference the two places with comments so that when someone else makes a change in one place they'll see that there is another that they also need to update.

Reply:
> If this cannot be avoided, then at least cross-reference the two places with comments so that when someone else makes a change in one place they'll see that there is another that they also need to update.

Done.

Reply:
> One is the CRD installed by the operator itself, the other is generated CRD installed explicitly when the operator installed via OperatorHub.

This still doesn't explain the "why": why do we have this the redundancy? Which of the two methods can be removed?

We shouldn't have two different ways of doing the same thing if it can be avoided. It's a source for bugs (see initial version of this PR...).

Reply:
Installing the CRD from inside the operator implies that it must have RBAC permission to do so. It seems simpler and more secure to me to remove the CRD creation from the operator.

Reply:
> Installing the CRD from inside the operator implies that it must have RBAC permission to do so. It seems simpler and more secure to me to remove the CRD creation from the operator.

Yes, we can drop the CRD creation. Let's do it in a separate PR/issue.

Reply:
Can we do that first and then rebase this one here? It makes no sense to introduce something here which will just be removed again.